src/v/storage/log_manager.cc

```diff
@@ -226,6 +226,18 @@ ss::future<log> log_manager::do_manage(ntp_config cfg) {
     });
 }
 
+ss::future<> log_manager::shutdown(model::ntp ntp) {
+    vlog(stlog.debug, "Asked to shutdown: {}", ntp);
+    return ss::with_gate(_open_gate, [this, ntp = std::move(ntp)] {
```

Comment:
There is a race condition here with the gate. You need to capture this exception and log error.

Reply:
What race condition do you have in mind ? This action isn't handled in background, caller will handle `gate_closed_exception`

Reply:
yeh I don't see any race condition here...